rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/recovery/NodesRecoveryManager.java

```diff
@@ -239,18 +263,40 @@ private void triggerDownNodeHookIfNecessary(NodeSource nodeSource, RMNodeData rm
     }
 
     private RMNode recoverNode(RMNodeData rmNodeData, NodeSource nodeSource,
-            Map<NodeState, Integer> recoveredNodeStatesCounter) {
+            Map<NodeState, Integer> recoveredNodeStatesCounter) throws InterruptedException {
         Node node = null;
         String nodeUrl = rmNodeData.getNodeUrl();
         NodeState previousState = rmNodeData.getState();
-        try {
-            node = nodeSource.lookupNode(nodeUrl, PAResourceManagerProperties.RM_NODELOOKUP_TIMEOUT.getValueAsInt());
-            logger.info("Node " + nodeUrl + " was looked up successfully");
-        } catch (Exception e) {
-            logger.info("Node " + nodeUrl + " could not be looked up");
+        boolean isPAMR = nodeUrl.startsWith("pamr://");
```

Comment:
Could you please extract constant here as well?

Reply:
ok